main.rb

```diff
@@ -1,44 +1,250 @@
-# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/ModuleLength, Metrics/MethodLength, Metrics/PerceivedComplexity, Lint/UselessAssignment, Metrics/CyclomaticComplexity, Metrics/BlockNesting, Style/IfInsideElse
 
 module Enumerable
-# 1.my_each
+  # 1.my_each
   def my_each
-    #return to_enum(:my_each) unless block_given?
+    # return to_enum(:my_each) unless block_given?
     if block_given?
       a = 0
-      while a < self.length
-        yield self[a]
+      while a < to_a.length
+        yield to_a[a]
         a += 1
       end
-      self 
-    else 
+      self
+    else
       to_enum(:my_each)
-    end  
+    end
   end
-  
-# 2.my_each_index 
+
+  # 2.my_each_index
   def my_each_with_index
-    #return to_enum(:my_each_index) unless block_given?
+    # return to_enum(:my_each_index) unless block_given?
     if block_given?
       a = 0
-      while a < self.length
-        yield(self[a], a)
+      while a < to_a.length
+        yield(to_a[a], a)
         a += 1
       end
-      self 
-    else 
+      self
+    else
       to_enum(:my_each_with_index)
-    end  
+    end
   end
 
-# 2.my_select
+  # 2.my_select
   def my_select
-    #return to_enum(:my_each_index) unless block_given?
+    # return to_enum(:my_each_index) unless block_given?
+    if block_given?
+      new_arr = []
+      to_a.my_each { |item| new_arr << item if yield(item) }
+      new_arr
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  # 4.my_all?
+  def my_all?(argument = nil)
+    index = 0
+    while index < length
+      if block_given?
+        return false if yield(self[index]) == false
+      else
+        if argument.nil? # when there is no argument
+          return false if self[index].nil? || self[index] == false
+        elsif argument.class == Regexp # when there is an argument
+          return false if self[index].match?(argument) != true
+        elsif argument.class == Class
+          return false if self[index].is_a?(argument) != true
+        else # when the argument is no regexp ni class
+          return false if self[index] != argument
+        end
+      end
+      index += 1
+    end
+    true
+  end
```

Comment:
- [ ] When `my_all?` is called on a range, it returns false if the block given is false.

```ruby
block = proc { |num| num > HIGHEST_VALUE } 
range = Range.new(5, 50)
range.my_all?(&block)
```
- Yours currently returns a `NameError: undefined local variable or method 'length' for 5..50:Range`